### ISSUE TITLE
多卡训练 Qwen3-Embedding-0.6B 的时候在 eval 阶段卡住不动

### ISSUE THREAD

**Issue by galabala** [modelscope/ms-swift#5810]
**Describe the bug**
使用H20 8卡训练卡死不动，train阶段没什么问题，到eval阶段就开始卡住不跑了，于是就CTRL+C了，输出如下：
![Image](https://github.com/user-attachments/assets/30ae3d6e-0848-427a-831a-a33073115dd6)

**Your hardware and system info**
NVIDIA-SMI 580.65.06              Driver Version: 580.65.06      CUDA Version: 13.0
卡住的时候GPU情况如下（这里举例一张）
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 580.65.06              Driver Version: 580.65.06      CUDA Version: 13.0     |
+-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA H20                     On  |   00000000:43:00.0 Off |                  Off |
| N/A   42C    P0            117W /  500W |   57245MiB /  97871MiB |    100%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A            1592      C   /root/miniconda3/bin/python           57236MiB |
+-----------------------------------------------------------------------------------------+

**Additional context**
执行命令（使用的是我自己数据集和loss）：
PYTHONPATH=/root/ConfitModel \
HF_HUB_CACHE=/root/autodl-tmp/raw_model \
HF_ENDPOINT=https://hf-mirror.com \
INFONCE_MASK_FAKE_NEGATIVE=true \
swift sft \
    --model Qwen/Qwen3-Embedding-0.6B \
    --task_type embedding \
    --model_type qwen3_emb \
    --train_type full \
    --dataset /root/autodl-tmp/train.parquet \
    --val_dataset /root/autodl-tmp/valid.parquet \
    --custom_dataset_info swift_training_data/dataset_info.json \
    --custom_register_path lib/lib_swift/__init__.py \
    --split_dataset_ratio 0.1 \
    --eval_strategy steps \
    --output_dir /root/autodl-tmp/swift_output/infonce-with-new-lr-lj-loss \
    --columns '{"job":"query","resume":"response"}' \
    --use_hf true \
    --logging_steps 2 \
    --save_steps 0.1 \
    --eval_steps 0.1 \
    --num_train_epochs 1 \
    --load_from_cache_file true \
    --max_length 4096 \
    --remove_unused_columns false \
    --save_total_limit 2 \
    --save_only_model true \
    --per_device_train_batch_size 12 \
    --per_device_eval_batch_size 12 \
    --gradient_accumulation_steps 4 \
    --learning_rate 6e-6 \
    --loss_type infonce_loss_and_multi_loss \
    --label_names labels \
    --dataloader_drop_last true

**Comment by slin000111** [comment]
> **Describe the bug** 使用H20 8卡训练卡死不动，train阶段没什么问题，到eval阶段就开始卡住不跑了，于是就CTRL+C了，输出如下： ![Image](https://github.com/user-attachments/assets/30ae3d6e-0848-427a-831a-a33073115dd6)
> 
> **Your hardware and system info** NVIDIA-SMI 580.65.06 Driver Version: 580.65.06 CUDA Version: 13.0 卡住的时候GPU情况如下（这里举例一张） +-----------------------------------------------------------------------------------------+ | NVIDIA-SMI 580.65.06 Driver Version: 580.65.06 CUDA Version: 13.0 | +-----------------------------------------+------------------------+----------------------+ | GPU Name Persistence-M | Bus-Id Disp.A | Volatile Uncorr. ECC | | Fan Temp Perf Pwr:Usage/Cap | Memory-Usage | GPU-Util Compute M. | | | | MIG M. | |=========================================+========================+======================| | 0 NVIDIA H20 On | 00000000:43:00.0 Off | Off | | N/A 42C P0 117W / 500W | 57245MiB / 97871MiB | 100% Default | | | | Disabled | +-----------------------------------------+------------------------+----------------------+
> 
> +-----------------------------------------------------------------------------------------+ | Processes: | | GPU GI CI PID Type Process name GPU Memory | | ID ID Usage | |=========================================================================================| | 0 N/A N/A 1592 C /root/miniconda3/bin/python 57236MiB | +-----------------------------------------------------------------------------------------+
> 
> **Additional context** 执行命令（使用的是我自己数据集和loss）： PYTHONPATH=/root/ConfitModel HF_HUB_CACHE=/root/autodl-tmp/raw_model HF_ENDPOINT=https://hf-mirror.com INFONCE_MASK_FAKE_NEGATIVE=true swift sft --model Qwen/Qwen3-Embedding-0.6B --task_type embedding --model_type qwen3_emb --train_type full --dataset /root/autodl-tmp/train.parquet --val_dataset /root/autodl-tmp/valid.parquet --custom_dataset_info swift_training_data/dataset_info.json --custom_register_path lib/lib_swift/**init**.py --split_dataset_ratio 0.1 --eval_strategy steps --output_dir /root/autodl-tmp/swift_output/infonce-with-new-lr-lj-loss --columns '{"job":"query","resume":"response"}' --use_hf true --logging_steps 2 --save_steps 0.1 --eval_steps 0.1 --num_train_epochs 1 --load_from_cache_file true --max_length 4096 --remove_unused_columns false --save_total_limit 2 --save_only_model true --per_device_train_batch_size 12 --per_device_eval_batch_size 12 --gradient_accumulation_steps 4 --learning_rate 6e-6 --loss_type infonce_loss_and_multi_loss --label_names labels --dataloader_drop_last true

看下卡哪儿了
pip install py-spy
py-spy dump --pid <pid>
